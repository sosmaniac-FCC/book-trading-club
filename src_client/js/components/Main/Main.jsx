import React from 'react';

/* global $ */

class Main extends React.Component {
    constructor(props) {
        super(props);
    }
    
    componentDidMount() {
        $(document).ready(() => {
            $("#option-4").trigger("click");
        });
    }
    
    render() {
        return (
            <div className="row" id="main">
                <div className="col s12">
                    <section className="col s12 m4">
                        <div className="icon-block">
                            <h2 className="center"><i className="material-icons">account_box</i></h2>
                            <h3 className="flow-text center">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
                            ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco 
                            laboris nisi ut aliquip ex ea commodo consequat.</h3>
                        </div>
                    </section>
                    <section className="col s12 m4">
                        <div className="icon-block">
                            <h2 className="center"><i className="material-icons">language</i></h2>
                            <h3 className="flow-text center">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
                            ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco 
                            laboris nisi ut aliquip ex ea commodo consequat.</h3>
                        </div>
                    </section>
                    <section className="col s12 m4">
                        <div className="icon-block">
                            <h2 className="center"><i className="material-icons">chat</i></h2>
                            <h3 className="flow-text center">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt 
                            ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco 
                            laboris nisi ut aliquip ex ea commodo consequat.</h3>
                        </div>
                    </section>
                </div>
            </div>
        );
    }
}

export default Main;