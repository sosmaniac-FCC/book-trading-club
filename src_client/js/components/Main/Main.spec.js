// expect from jest_expect.js
/* global expect */

import React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeAdapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

import Main from './Main.jsx';

configure({ adapter: new enzymeAdapter() });

test('Main page (home route) content loads as expected.', () => {
    const component = shallow(<Main />);
    const tree = toJSON(component);
    expect(tree).toMatchSnapshot();
});

test('Main page (home route) content should have no style attributes.', () => {
    const component = shallow(<Main />);
    expect(component.find('[style]')).toHaveLength(0);
});