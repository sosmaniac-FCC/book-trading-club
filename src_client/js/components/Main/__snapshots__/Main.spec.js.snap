// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page (home route) content loads as expected. 1`] = `
<div
  className="row"
  id="main"
>
  <div
    className="col s12"
  >
    <section
      className="col s12 m4"
    >
      <div
        className="icon-block"
      >
        <h2
          className="center"
        >
          <i
            className="material-icons"
          >
            account_box
          </i>
        </h2>
        <h3
          className="flow-text center"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </h3>
      </div>
    </section>
    <section
      className="col s12 m4"
    >
      <div
        className="icon-block"
      >
        <h2
          className="center"
        >
          <i
            className="material-icons"
          >
            language
          </i>
        </h2>
        <h3
          className="flow-text center"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </h3>
      </div>
    </section>
    <section
      className="col s12 m4"
    >
      <div
        className="icon-block"
      >
        <h2
          className="center"
        >
          <i
            className="material-icons"
          >
            chat
          </i>
        </h2>
        <h3
          className="flow-text center"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </h3>
      </div>
    </section>
  </div>
</div>
`;
